src/components/services/SortedWalletList.js

```diff
@@ -0,0 +1,297 @@
+        token,
+        tokenId,
+        wallet,
+        walletId
+      })
+    }
+  }
+
+  // Phase 2: Sort the list.
+  let sorted = wallets
+  switch (walletsSort) {
+    case 'currencyCode':
+      sorted = stableSort(
+        wallets,
+        alphabeticalSort(({ token, wallet }) => {
+          if (token != null) return token.currencyCode
+          if (wallet != null) return wallet.currencyInfo.currencyCode
+        })
+      )
+      break
+
+    case 'currencyName':
+      sorted = stableSort(
+        wallets,
+        alphabeticalSort(({ token, wallet }) => {
+          if (token != null) return token.displayName
+          if (wallet != null) return wallet.currencyInfo.displayName
+        })
+      )
+      break
+
+    case 'highest':
+      sorted = stableSort(
+        wallets,
+        numericSort(item => {
+          const value = getFiat(item, exchangeRates)
+          if (value != null) return -value
+        })
+      )
+      break
+
+    case 'lowest':
+      sorted = stableSort(
+        wallets,
+        numericSort(item => getFiat(item, exchangeRates))
+      )
+      break
+
+    case 'name':
+      sorted = stableSort(
+        wallets,
+        alphabeticalSort(({ wallet }) => {
+          if (wallet == null) return
+          if (wallet.name != null) return wallet.name
+          return sprintf(s.strings.my_crypto_wallet_name, wallet.currencyInfo.displayName)
+        })
+      )
+      break
+  }
+
+  // Phase 3: Check for differences.
+  const dispatch = useDispatch()
+  const emptyList: WalletListItem[] = [] // Needed for Flow.
+  const lastList = useRef(emptyList)
+  useEffect(() => {
+    if (!matchWalletList(sorted, lastList.current)) {
+      dispatch({ type: 'UPDATE_SORTED_WALLET_LIST', data: sorted })
+    }
+    lastList.current = sorted
+  })
+
+  return null
+}
+
+/**
+ * Subscribes to all the tokens across all plugins in an account.
+ */
+export function useAllTokens(account: EdgeAccount): EdgeTokenMaps {
+  // Gather the tokens from all the plugins:
+  const [out, setOut] = useState<EdgeTokenMaps>(() => {
+    const out: EdgeTokenMaps = {}
+    for (const pluginId of Object.keys(account.currencyConfig)) {
+      out[pluginId] = account.currencyConfig[pluginId].allTokens
+    }
+    return out
+  })
+
+  // Watch for token changes on any of the plugins:
+  useEffect(() => {
+    const cleanups: Array<() => void> = []
+
+    for (const pluginId of Object.keys(account.currencyConfig)) {
+      const currencyConfig = account.currencyConfig[pluginId]
+      function update(): void {
+        setOut(out => ({ ...out, [pluginId]: currencyConfig.allTokens }))
+      }
+      cleanups.push(currencyConfig.watch('allTokens', update))
+    }
+
+    return () => cleanups.forEach(cleanup => cleanup())
+  }, [account])
+
+  return out
+}
+
+/**
+ * Subscribes to all the enabled token lists in the account.
+ */
+function useEnabledWalletIds(account: EdgeAccount): EnabledTokenIds {
```

Comment:
Can we use useWatchAccount for this and useAllTokens?

Reply:
Neither of these are simple properties - we need multiple subscriptions, one per item, so we have to build a bit fancier hook.